### DISCUSSION TITLE
Auto-rotation moves light too

### DISCUSSION THREAD

**Discussion by tom-deadline** [google/model-viewer#2369] [Q&A] [answered]
When auto-rotate is enabled during rotation, light moves with the model, and when the user spins the model after auto-rotation, light is behind the model and stays there no matter how you spin the model.

**Comment by elalish** [comment]
Correct. Note this example: https://modelviewer.dev/examples/stagingandcameras/#turnSkybox

**Comment by tom-deadline** [comment, marked as answer]
is there a way to make the light position permanent, so on auto-rotate it will stay in its initial position?

**Reply by elalish** [reply]
Wait, I think I misread your first comment. With `auto-rotate` the light is fixed; only the model moves. If you want the light and the object to stay in the same relative position, then you'll want to set new `camera-orbit` values in `requestAnimationFrame()` rather than using `auto-rotate`.

**Reply by tom-deadline** [reply]
That's the problem, with auto-rotation light moves, it is not fixed

**Reply by elalish** [reply]
I think we might be having a disagreement of reference spaces 😂. Can you post a video or repro link that shows the problem you're seeing and describe what you expect? It might also be easier to see what's going on if you set a `skybox-image` so that the lights are visible in the background.

**Reply by tom-deadline** [reply]
https://drive.google.com/file/d/1d5lzr0xt2coe-t66OpNdSHaB-5bPNtjf/view?usp=sharing

here's a video, during auto-rotate and user interaction the light behaves itself differently.

**Reply by elalish** [reply]
Yes, that's correct and intentional. User interaction keeps the lights and object locked together (the user is moving themselves around the static object). Auto-rotate rotates the object relative to the lights (and background if you have one) while the user stays fixed. So, it sounds like instead of using auto-rotate, you'd rather orbit using the camera-orbit attribute, which you can set inside of a requestAnimationFrame().

**Reply by tom-deadline** [reply]
got it, thanks
